frontend/src/methods/auth.ts

```diff
@@ -237,3 +241,29 @@ const updateToken = async (tokenID: string, update: (token: Resource<JoinTokenSp
 
   await ResourceService.Update(token, token.metadata.version, withRuntime(Runtime.Omni))
 }
+
+const redirectToURL = (url: string) => {
+  if (window.top) {
+    window.top.location.href = url
+  } else {
+    window.location.href = url
+  }
+}
+
+export const logout = async (auth0: Auth0VueClient | undefined) => {
+  await revokePublicKey()
+
+  await auth0?.logout({
+    logoutParams: {
+      returnTo: window.location.origin,
+    },
+  })
+
+  resetKeys()
+
+  currentUser.value = undefined
+
+  if (authType.value !== AuthType.Auth0) {
```

Comment:
wasnt it before `(authType.value === AuthType.OIDC)`? and `SAML` did `navigateToLogin`